components/eam/src/physics/cam/phys_grid_ctem.F90

```diff
@@ -269,44 +276,37 @@ subroutine phys_grid_ctem_diags(phys_state)
    real(r8) :: vthza(nzalat,pver)
    real(r8) :: wthza(nzalat,pver)
 
+   real(r8) :: psza(nzalat)
    real(r8) :: uza(nzalat,pver)
    real(r8) :: vza(nzalat,pver)
    real(r8) :: wza(nzalat,pver)
    real(r8) :: thza(nzalat,pver)
 
-   real(r8) :: mbarv ! molecular weight of dry air (g/mol)
-   real(r8) :: sheight(pcols,pver) ! pressure scale height (m)
-
-   if (.not.do_calc()) return
-
    ! In CESM/WACCM the variable mbarv is provided by the "air_composition" 
    ! module, which is not in E3SM, so we just use a rough approximation
-   mbarv = 28.97
+   real(r8) :: mbarv = 28.97 ! molecular weight of dry air (g/mol)
```

Comment:
Molecular weight of dry air is `mwdair`in `components/eam/src/control/physconst.F90`, and can be traced back to `share/util/shr_const_mod.F90`:
` real(R8),parameter :: SHR_CONST_MWDAIR  = 28.966_R8       ! molecular weight dry air ~ kg/kmole`

Reply:
oh, I didn't realize we had a constant for this. I'll update to use the shared value.

Reply:
actually, I forgot that we can't delete this entirely since we don't need the scale height anymore.

Reply:
pushed a commit to remove this.